MetaDataset.py

```diff
@@ -958,6 +1040,17 @@ def _expand_dataset_sec_idxs(self, num_values):
               self.estimated_num_seq_per_subset[dataset_idx] = self.used_num_seqs_per_subset[dataset_idx]
 
       else:
+        # If we don't need to expand it is ok if the sorting method is not supported.
```

Comment:
Why is that relevant? Why not always throw the exception? Wouldn't you run always into cases where it needs to be expanded at some point?

Reply:
Ah this is because you always call `_expand_dataset_sec_idxs` now?
What `seq_ordering` settings does this `else` branch cover here exactly now? For those `seq_ordering` cases where you know that you must have used `num_seqs` in advance, you might directly `return False`. And all other cases should already be handled above, right?

Reply:
Yes, this is only for the sequence orderings implemented in `get_seq_order_for_epoch()`, i.e. the implementation I use when `num_seqs` is known. But do I really want to check here if the sequence ordering is one of those? (random, sorted, laplace, sort_bin_shuffle...). A check whether we know `num_seqs` and therefore used `get_seq_order_for_epoch()` in the first place would be more intuitive to me. (That's what I said I don't like :D)

Reply:
I see. I think the current solution is still the easiest one and shortest one.
You might want to extend this comment though like this:
```suggestion
        # If we don't need to expand it is ok if the sorting method is not supported.
        # This can happen if we already sorted everything in advance in `init_seq_order`.
```